dspace-oai/src/main/java/org/dspace/event/OAIIndexEventConsumer.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.event;
+
+import java.sql.SQLException;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import org.apache.log4j.Logger;
+import org.dspace.content.Bitstream;
+import org.dspace.content.Bundle;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.DSpaceObject;
+import org.dspace.content.Item;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Constants;
+import org.dspace.core.Context;
+import org.dspace.xoai.app.BasicConfiguration;
+import org.dspace.xoai.app.XOAI;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+public class OAIIndexEventConsumer implements Consumer {
+    /**
+     * log4j logger
+     */
+    private static Logger log = Logger.getLogger(OAIIndexEventConsumer.class);
+
+    ItemService itemService = ContentServiceFactory.getInstance().getItemService();
+    // collect Items, Collections, Communities that need indexing
+    private Set<Item> itemsToUpdate = null;
+
+    public void initialize() throws Exception {
+        // No-op
+    }
+
+    /**
+     * Consume a content event -- just build the sets of objects to add (new) to
+     * the index, update, and delete.
+     *
+     * @param ctx   DSpace context
+     * @param event Content event
+     */
+    public void consume(Context ctx, Event event) throws Exception {
+
+        if (itemsToUpdate == null) {
+            itemsToUpdate = new HashSet<Item>();
+        }
+
+        int st = event.getSubjectType();
+        if (!(st == Constants.ITEM || st == Constants.BUNDLE
+                || st == Constants.COLLECTION || st == Constants.COMMUNITY || st == Constants.BITSTREAM)) {
+            log
+                    .warn("IndexConsumer should not have been given this kind of Subject in an event, skipping: "
+                            + event.toString());
+            return;
+        }
+
+        DSpaceObject subject = event.getSubject(ctx);
+        DSpaceObject object = event.getObject(ctx);
+
+
+        int et = event.getEventType();
+
+        if (object != null && event.getObjectType() == Constants.ITEM) {
+            //update just object
+            itemsToUpdate.add((Item)object);
+            return;
+        }
+
+        if (subject != null) {
+            if (event.getSubjectType() == Constants.COLLECTION || event.getSubjectType() == Constants.COMMUNITY) {
```

Comment:
So many if, else statements, it is really hard to read, is it possible to refactor it?

Reply:
This code is from v5. I left it as is because, in this project, it’s not the only class I copied and pasted from v5.